omega/omega-transport/omega-transport-resttemplate/src/main/java/org/apache/servicecomb/saga/omega/transport/resttemplate/TransactionClientHttpRequestInterceptor.java

```diff
@@ -47,7 +52,7 @@ private String globalTxId() {
     String globalTxId = omegaContext.globalTxId();
 
     if (globalTxId == null) {
-      return omegaContext.newGlobalTxId();
+      LOG.error("Global tx id should not be null.");
```

Comment:
this message is misleading, since the rest template may be used for non transaction request. better just not setting header at all and show error at compensable aspect.

Reply:
Not setting the header when it's null, but we still need to inject the header otherwise. Just not generate the id here, right?

Reply:
user is supposed to add @SagaStart to generate global tx id